core/app/models/spree/order_promotion.rb

```diff
@@ -4,7 +4,7 @@ class OrderPromotion < Spree::Base
     belongs_to :promotion, class_name: 'Spree::Promotion'
 
     delegate :name, :description, :code, to: :promotion
-    delegate :currency, to: :order
+    delegate :currency, :public_metadata, to: :order
```

Comment:
@aleksandarpetrushev let's remove this as this would be duplication

Reply:
Issue arises when trying to serialize the Order (Cart). Namely it tries to
serialize OrderPromotion when promotions is included, and when doing that
it can not find the .public_metadata, so we need to delegate either to order
or promotion. Maybe we should have an OrderPromotion serializer like for Platform
API.

Reply:
It should use Promotion public_metadata in that case